File: Java Script - Introduction to Programming/8.1. Exam Preparation - Part I/08. Increasing Elements.js
function increasingElements(arg) {
  arg = arg.map(Number)
  let countCurrentLongest = 0
  let countLongest = 0
  let numPrev = 0
  let current = 0

  for (let i = 1; i < arg.length; i++) {
      current = arg[i]

      if(current > numPrev) {
          countCurrentLongest++
      } else {
          countCurrentLongest = 1
      }

      if(countCurrentLongest > countLongest) {
          countLongest = countCurrentLongest
      }

      numPrev = current
  }

  console.log(countLongest)
}

// increasingElements(['5', '2', '4'])
//increasingElements(['4','2', '8', '7', '6'])
//increasingElements(['4', '1', '2', '4', '4'])
// increasingElements(['4','5', '6', '7', '8'])

File: Pyrhon - Introduction to Programming/8.1. Exam Preparation - Part I/08. Increasing Elements.py
loop = int(input())
arg = [float(input()) for _ in range(loop)]
count_current_longest = 0
count_longest = 0
num_prev = 0


for i in range(0, len(arg)):
    current = arg[i]

    if current > num_prev:
        count_current_longest+= 1
    else:
        count_current_longest = 1

    if count_current_longest > count_longest:
        count_longest = count_current_longest

    num_prev = current


print(count_longest)


'''
3
5
2
4
#2
'''
